### PR TITLE
Update linting, TypeScript, and Jest

### PR DESCRIPTION
**Pull request by arsalansufi**
# Overview

This PR makes a few updates to the template-node-library, to match it up with the recently updated [template-frontend](https://github.com/Clever/template-frontend). The updates include:

- The latest linting packages and config
- The latest TypeScript
- The latest testing packages

# Testing

Verified that all of the following work as expected:

- [x] `make format` / `make format-all`
- [x] `make lint`
- [x] `make test`
- [x] `make build`

# Rollout

:100:

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+module.exports = {
```
Comment (arsalansufi) [inline .eslintrc.js:1]: An exact copy of https://github.com/Clever/template-frontend/blob/master/.eslintrc.js. We eventually plan to move this config to a shared location.

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
```
Comment (arsalansufi) [inline jest.config.js:1]: Essentially the same config as https://github.com/Clever/template-frontend/blob/master/jest.config.js

File: .eslintrc.js (hunk 1 above)
Comment (arsalansufi) [inline .eslintrc.js:1]: An exact copy of https://github.com/Clever/template-frontend/blob/master/.eslintrc.js. We eventually plan to move this config to a shared location.

File: tsconfig.json [hunk 3]
```diff
@@ -1,15 +1,13 @@
...
+    "jsx": "react",
+    "lib": ["dom", "es2019"],
+    "module": "commonjs",
+    "outDir": "dist",
+    "resolveJsonModule": true,
+    "strict": true,
+    "target": "es6",
+    "typeRoots": ["node_modules/@types", "types"]
+  },
+  "exclude": ["node_modules"]
```
Comment (arsalansufi) [inline tsconfig.json:12]: Some small tweaks to better mirror https://github.com/Clever/template-frontend/blob/master/tsconfig.json

File: jest.config.js (hunk 2 above)
Comment (arsalansufi) [inline jest.config.js:1]: Essentially the same config as https://github.com/Clever/template-frontend/blob/master/jest.config.js

File: tsconfig.json [hunk 4]
```diff
@@ -1,15 +1,13 @@
...
-        "node_modules"
-    ]
+  "compilerOptions": {
+    "jsx": "react",
+    "lib": ["dom", "es2019"],
+    "module": "commonjs",
+    "outDir": "dist",
+    "resolveJsonModule": true,
+    "strict": true,
+    "target": "es6",
```
Comment (arsalansufi) [inline tsconfig.json:9]: I want to do some additional thinking about the target setting.

For frontend code builds, we target es5 for compatibility with IE11 and older versions of Safari, e.g. [here](https://github.com/Clever/template-frontend/blob/5f4062967815dfbd898c7a3c827de86551e0e8ab/tsconfig.client.json#L4). For server code on the other hand, it's okay to set a higher target, e.g. [here](https://github.com/Clever/template-frontend/blob/5f4062967815dfbd898c7a3c827de86551e0e8ab/tsconfig.json#L12).

Given that this template can be used to create npm modules for server use or client use, I'm wondering if the safest default is es5. For someone creating a server module, they can then explicitly choose to raise the target.

Comment (bgveenstra) [reply tsconfig.json:9]: I "like" ES5 as the default while we're supporting IE11. Could you add instructions to `Development` in the README for changing the target and when it's appropriate?

**Review by bgveenstra — APPROVED** [review]
LGTM.   Whether the template tsc target is ES5 or ES6, I think a line about it in the readme (why&how to change) would be very helpful.

File: tsconfig.json (hunk 4 above)
Comment (arsalansufi) [reply tsconfig.json:9]: Yes will definitely add!

**Pull request merged**